fuel-txpool/src/txpool.rs

```diff
@@ -145,11 +150,53 @@ impl TxPool {
         Ok(())
     }
 
+    #[cfg(feature = "p2p")]
+    pub async fn insert_with_broadcast(
+        txpool: &RwLock<Self>,
+        db: &dyn TxPoolDb,
+        tx_status_sender: broadcast::Sender<TxStatusBroadcast>,
+        network_sender: mpsc::Sender<P2pRequestEvent>,
+        txs: Vec<ArcTx>,
+    ) -> Vec<anyhow::Result<Vec<ArcTx>>> {
+        let mut res = Vec::new();
+        for tx in txs.iter() {
+            let mut pool = txpool.write().await;
+            res.push(pool.insert_inner(tx.clone(), db).await)
+        }
+        for (ret, tx) in res.iter().zip(txs.into_iter()) {
+            match ret {
+                Ok(removed) => {
```

Comment:
In the current code, we check that transactions were in the transactions pool, but we don't check that transactions have been committed in the block.

With the introduction of P2P gossiping seems we need to do this check. Overwise, we will gossip about already committed transactions)

Reply:
I think txpool cleanup should be deferred to the block importer/sync task. Without that in place, it doesn't make as much to implement this yet.